python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -448,9 +457,31 @@ def do_evaluate(
         skip_rows: int,
         n_rows: int,
         row_index: tuple[str, int] | None,
+        include_file_paths: str | None,
         predicate: expr.NamedExpr | None,
     ) -> DataFrame:
         """Evaluate and return a dataframe."""
+
+        def _include_file_paths_column(
+            include_file_paths: str | None, df: DataFrame
+        ) -> DataFrame:
+            plc_col = plc.interop.from_arrow(
+                pa.array(
+                    [
+                        [
+                            s
+                            for s, n in zip(
+                                paths, tbl_w_meta.num_rows_per_source, strict=False
+                            )
+                            for _ in range(n)
+                        ]
+                    ]
+                )
+            )
+            (_, col) = plc_col.children()
+            paths_col = Column(col, name=include_file_paths)
+            return df.with_columns([paths_col])
```

Comment:
These columns could be very big, and we're producing them by iterating in python.

What we need to do for each separate file is to remember the name of the path we read, and how many rows that file produced. We can then use `plc.filling.repeat` to make something of the right shape.

Suppose we gather `paths: list[str]` and `repeats : list[int]` then:

```
(filepaths,) = plc.filling.repeat(
    plc.Table([plc.interop.from_arrow(pa.array(paths))]),
    plc.interop.from_arrow(pa.array(repeats, type=pa.int32())),
).columns()
```